### PR TITLE
Load test with k6

### PR DESCRIPTION
**Pull request by QYuQianchen**
This PR closes https://github.com/Rpc-h/RPCh/issues/510

### Changes
- Add load testing for 'rpc-server', to simulate close-to-real-world traffics sent by wallet.
- Turn the legacy implementation into a modular implementation. Remove all the existing implementation, except for the k6 "test options" definition
- Add new test options. Available options are: SOAK, SMOKE, SPIKE, STRESS, LOAD, BURST, CONSTANT, LONG
- Create "MockWallet" class to enable wallet-centric testing. Available mocked wallets: DUMMY, METAMASK, OKX, RABBY, FRAME
- Create real-world traffic, extracted mainly from DEXes (Uniswap and CoWSwap) as they are very representative for heavy[ JSON-RPC](https://ethereum.github.io/execution-apis/api-documentation/) calls. 
- Enhance documentation: Add [README](https://github.com/Rpc-h/RPCh/blob/b3e9f5c56a1f6c50978be41e38b2b72810f6c068/DEVELOPER_SETUP.md) for launching a local RPCh node; Improve [readme](https://github.com/Rpc-h/RPCh/blob/b3e9f5c56a1f6c50978be41e38b2b72810f6c068/devkit/loadtesting-k6/README.md) for load-testing (e.g. how to extend for more wallets and include typical traffic for other dApps)

### Miscellaneous
- add lint-ignore for `runOneHops` in `RPCh/apps/availability-monitor` to make the build pass

## Test
_As specified in the README file_
1. Spin up a RPCh node of staging environment where the RPC Server runs on http://localhost:45740/. Please refer to DEVELOPER_SETUP.md for more details
2. Navigate to `devkit/loadtesting-k6`
3. Run:
```bash
export TEST_TYPE=SMOKE WALLET_TYPE=METAMASK ; \
   RPC_SERVER_URL=http://localhost:45750 \
   k6 run ./dist/rpc-server/index.js \
   --out json="results/test-$TEST_TYPE-$WALLET_TYPE-$(date +%s).json"
```
Where a list of values for `TEST_TYPE` and `WALLET_TYPE` can be found with:
```
yarn run help
```

4. Check the result json file and "500” means that there’s failure in RPCh side

### CODE REVIEW

File: devkit/loadtesting-k6/README.md [hunk 1]
```diff
@@ -5,30 +5,71 @@
...
+## Test
+### How to test RPC Server
+
+1. Spin up Sandbox or all the infrastracture locally (except for the entry/exit nodes) or spin up a RPCh node of staging environment where the RPC Server runs on http://localhost:45740/. Please refer to [DEVELOPER_SETUP](../../DEVELOPER_SETUP.md) for more details
+2. Navigate to `devkit/loadtesting-k6`
+3. Run:
+```bash
+export TEST_TYPE=SMOKE WALLET_TYPE=METAMASK ; \
+   RPC_SERVER_URL=http://localhost:45750 \
+   k6 run ./dist/rpc-server/index.js \
```
Comment (esterlus) [inline devkit/loadtesting-k6/README.md:24]: should be `yarn exec` to avoid calling global binaries

**Review by esterlus — APPROVED** [review]
Nice work. Please update README as per my comment before landing.

File: devkit/loadtesting-k6/README.md (hunk 1 above)
Comment (QYuQianchen) [reply devkit/loadtesting-k6/README.md:24]: Just noticed that `"k6"` shouldn't appear in the `package.json`. For this command, it's expected to call the (global) installed binary of "k6"

Comment (esterlus) [reply devkit/loadtesting-k6/README.md:24]: Any reason why this should be a global dependency?

Comment (QYuQianchen) [reply devkit/loadtesting-k6/README.md:24]: Given `yarn exec` restricts the execution environment to current workspace, the assumption that it's in the current workspace that users have installed "k6"  should hold for the command to be executable. As we do not provide script for installing k6 under the project's folder, the command may fail when it's wrapped with `yarn exec`.

Comment (esterlus) [reply devkit/loadtesting-k6/README.md:24]: It's a dependency in the `package.json`, isn't it?

Comment (QYuQianchen) [reply devkit/loadtesting-k6/README.md:24]: As per DM, it's resolved

**Pull request merged**